src/Package/Impl/Repl/VsRInteractiveWorkflowProvider.cs

```diff
@@ -74,10 +79,20 @@ internal class VsRInteractiveWorkflowProvider : IRInteractiveWorkflowProvider, I
                                                     _plotsProvider, _activeTextViewTracker, _debuggerModeTracker, 
                                                     _shell, settings, _wss, () => DisposeInstance(sessionProvider));
             _disposableBag.Add(workflow);
+
+            sessionProvider.BrokerChanging += OnBrokerChanging;
             return workflow;
         }
 
+        private void OnBrokerChanging(object sender, EventArgs e) {
```

Comment:
Ok, we have 5 places where we call `GetOrCreateVisualComponent`, and in 4 of them we then show it. So I think we can just add a couple of parameters `(bool showWindow = false, bool focus = false)` to the `GetOrCreateVisualComponent` and move code that shows the window right there.

---
In reply to: [79260440](https://github.com/Microsoft/RTVS/pull/2341#discussion_r79260440) [](ancestors = 79260440)